### PR TITLE
Refactor to enable start in user supervision tree

### PR DESCRIPTION
**Pull request by fhunleth**
This moves quite a bit of code around, but the change is pretty small
since the application environment was previously being passed as
arguments to the daemon GenServer anyway. Most of the change was adding
the `case` to automatically start the daemon if `:nerves_ssh` was in the
application environment. If it's not, then nothing happens. This seems
quite reasonable since if you didn't specify anything in the application
config previously, the SSH daemon would have started, but you couldn't
log in.

The rest of the changes are adjusting unit tests and documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,9 +54,27 @@ config :shoehorn,
 your main OTP application stops, OTP may stop `:nerves_ssh`, and that would make
 your device inaccessible via SSH.
 
+### Starting as part of one of your supervision trees
+
+If you want to do this, make sure that you do NOT specify `:nerves_ssh` in your
+`config.exs`. The `:nerves_ssh` decides whether or not to automatically launch
```
Comment (jjcarstens) [inline README.md:60]:
```suggestion
`config.exs`. The `:nerves_ssh` key decides whether or not to automatically launch
```

File: README.md [hunk 2]
```diff
@@ -39,9 +54,27 @@ config :shoehorn,
...
+
+Then when specifying the children for your supervisor, add `NervesSSH` like
+this:
+
+```elixir
+    {NervesSSH, nerves_ssh_options}
+```
+
+The `nerves_ssh_options` should be a `NervesSSH.Options` struct. See the
+`Configuration` section things that you may specify. Calling
```
Comment (jjcarstens) [inline README.md:71]:
```suggestion
`Configuration` section option fields that you may specify. Calling
```

**Comment by fhunleth** [comment]
@jjcarstens Thanks - I updated with all of your suggested changes!

**Comment by fhunleth** [comment]
Oops - I pushed changes up in the wrong order and GitHub thinks I closed it without merging. I merged.

**Pull request closed without merging**